Create class AnimalPhoto

AlexeyVolkov2023 opened the pull request:
Создал класс AnimalPhoto, заменил String.Empty на default!  в классах Animal, Volunteer . Создал фабричный метод для классов HelpDetails и SocialNetworks

AlexeyVolkov2023 committed:
Create class AnimalPhoto

Создал класс AnimalPhoto, заменил String.Empty на default!  в классах Animal, Volunteer . Создал фабричный метод для классов HelpDetails и SocialNetworks

KirillSachkov commented on backend/src/HelpAnimal.Domain/Models/AnimalPhoto.cs:
```diff
@@ -0,0 +1,18 @@
+    public string StoragePath { get; private set; } = default!;
+    public bool IsMain { get; private set; }
+
+    public static AnimalPhoto Create(string storagePath, bool isMain)
```
Нужно сделать приватный конструктор и вызывать его в методе Create

KirillSachkov commented on backend/src/HelpAnimal.Domain/Models/Animal.cs:
```diff
@@ -3,30 +3,32 @@
+    public List<AnimalPhoto> Photos { get; private set; } = []; // Список фотографий
 
     
     public static Animal Create (string name, string species, string description,
```
Нужно сделать приватный конструктор и вызывать его в методе Create

KirillSachkov commented on backend/src/HelpAnimal.Domain/Models/HelpDetails.cs:
```diff
@@ -5,9 +5,12 @@ public class HelpDetails
     {
-        Requisite = requisite;
-        Description = description;
+        return new HelpDetails()
```
Через приватный конструктор

KirillSachkov commented on backend/src/HelpAnimal.Domain/Models/SocialNetwork.cs:
```diff
@@ -5,9 +5,12 @@ public class SocialNetwork
     public string Link { get;private set; } // Ссылка на профиль
 
-    public SocialNetwork(string name, string link)
+    public static SocialNetwork Create(string name, string link)
```
Через приватный конструктор

AlexeyVolkov2023 committed:
Create private constructor for classes

Создал приватные конструкторы для классов Animal, Volunteer, HelpDetails, Petphoto, SocialNetwork

AlexeyVolkov2023 commented:
Где найти информацию по поводу создания EFCore  Id,  почему не надо задавать его в конструкторе класса?

KirillSachkov commented:
Ef Core сам генерирует Id, когда сохраняет сущность в базу данных. Я ещё расскажу про это
